kube-runtime/src/watcher.rs

```diff
@@ -225,15 +233,28 @@ async fn step<K: Resource + Clone + DeserializeOwned + Debug + Send + 'static>(
 /// that we have seen on the stream. If this is successful then the stream is simply resumed from where it left off.
 /// If this fails because the resource version is no longer valid then we start over with a new stream, starting with
 /// an [`Event::Restarted`].
-pub fn watcher<K: Resource + Clone + DeserializeOwned + Debug + Send + 'static>(
+pub fn watcher<K: Resource + Clone + DeserializeOwned + Debug + Send + 'static, B: Backoff + Send>(
```

Comment:
Overall the backoff logic is basically independent from the watcher itself. I'd much rather put that in a generic wrapper in `utils`. Especially if you want the `Observer` wrapper to be the "simple" entry point that users use.

Reply:
have made a stream wrapper, but it was really hard to make it do what i wanted :(

there's a bunch of commented out attempts using `stream::scan`, but all of them have awful closure lifetime issues that i could not figure out, so have taken the async-stream shortcut for now. not sure if this is smart, or if you see a way that i can avoid this. have left most of my attempts in there commented out, but at least the async-stream variant works.

Reply:
I think this is actually a good fit for a manual `enum` state machine, but I'll take a closer look.